### PR TITLE
Loki: Pass log volume hint as a request header

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

`hint` param is a meta data and should not be passed with request params in the URL. @grafana/loki-team decided it should be a request header.

**Which issue(s) this PR fixes**:

Fixes N/A

**Special notes for your reviewer**:

**How to test it?**

`hint` param should no longer be passed and new request header should be added, but only to log volume query:

| regular query | log volume query |
|--------------|-------------------|
|  Screen Shot 2021-11-23 at 09 17 07 | Screen Shot 2021-11-23 at 09 18 12 |

CC: @slim-bean @kavirajk

### CODE REVIEW

**Review by kavirajk — APPROVED** [review]
LGTM! Thanks @ifrost

**Review by Elfo404 — APPROVED** [review]
LGTM! 🚀

File: public/app/plugins/datasource/loki/datasource.test.ts [hunk 1]
```diff
@@ -184,14 +184,26 @@ describe('LokiDatasource', () => {
 
       it('should add volume hint param for log volume queries', () => {
         const target = { expr: '{job="grafana"}', refId: 'B', volumeQuery: true };
-        const req = ds.createRangeQuery(target, options as any, 1000);
-        expect(req.hint).toBe('logvolhist');
+        ds.runRangeQuery(target, options);
+        expect(backendSrv.fetch).toBeCalledWith(
+          expect.objectContaining({
+            headers: {
+              'X-Query-Tag': 'Source=logvolhist',
```
Comment (cyriltovena) [inline public/app/plugins/datasource/loki/datasource.test.ts:191]: Wasn't it supposed to be `X-Query-Tags` I guess it's not a big deal.

Comment (ifrost) [reply public/app/plugins/datasource/loki/datasource.test.ts:191]: Fixed in https://github.com/grafana/grafana/pull/42205

**Pull request merged**